packages/extension-base/src/services/fee-service/utils/index.ts

```diff
@@ -216,6 +219,21 @@ export const calculateGasFeeParams = async (web3: _EvmApi, networkKey: string, u
       };
     }
 
+    if (networkKey === 'polygonzkEvm_cardona' || networkKey === 'polygonZkEvm') {
+      const isTestnet = networkKey === 'polygonzkEvm_cardona';
```

Comment:
Is it necessary to use gas indexer of polygon?
Maybe use the fallback of this function?

Reply:
Move this code to the header of try